codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -233,17 +234,8 @@ class ServerProtocolTestGenerator(
         httpRequestTestCase.host.orNull()?.also {
             rust("""todo!("endpoint trait not supported yet");""")
         }
-        checkQueryParams(this, httpRequestTestCase.queryParams)
-        checkForbidQueryParams(this, httpRequestTestCase.forbidQueryParams)
-        checkRequiredQueryParams(this, httpRequestTestCase.requireQueryParams)
-        checkHeaders(this, "&http_request.headers()", httpRequestTestCase.headers)
-        checkForbidHeaders(this, "&http_request.headers()", httpRequestTestCase.forbidHeaders)
-        checkRequiredHeaders(this, "&http_request.headers()", httpRequestTestCase.requireHeaders)
         if (protocolSupport.requestBodyDeserialization) {
-            // "If no request body is defined, then no assertions are made about the body of the message."
-            httpRequestTestCase.body.orNull()?.also { body ->
-                checkBody(this, body, httpRequestTestCase)
-            }
```

Comment:
This is what I was talking about when I wanted to remove the client behaviour!! Thanks!